src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -3244,6 +3220,26 @@ public bool TryGetCoercionSubType(DType expectedType, out DType coercionType, ou
                 return CoercesTo(expectedType, out var coercionIsSafe, aggregateCoercion) && (!safeCoercionRequired || coercionIsSafe);
             }
 
+            // LazyTable/Record case
+            if (expectedType.IsLazyType)
+            {
+                coercionType = expectedType;
+                coercionNeeded = !expectedType.Accepts(this);
+                if (!coercionNeeded)
+                {
+                    return true;
+                }
+                
+                if (expectedType.LazyTypeProvider.GetExpandedType(expectedType.IsTable).Accepts(this))
```

Comment:
>expectedType.IsTable [](http://example.com/codeflow?start=66&length=20)

It's little odd that this information is needed to be passed here. Shouldn't lazy type provider of expectedType already have this information?

Why BackingFormulaType not have this information?

Reply:
So, LazyTypes can be either tables or records, with the same backing formula type. this ensures that we maintain that context when we expand the type fully.

Reply:
At a time, LazyType can be either table or record, right? In that case BackingFormulaType should have this context.

Reply:
The BackingFormulaType is always a RecordValue. 
The DType that wraps it can be either a table or record.
This allows us to call `ToRecord()` and `ToTable()` without affecting the backing type at all, and ensures that identity properties around Accepts() work correctly. 

When we hit a point where we need to fully expand the type, we need to know whether to expand it as a table, or as a record.